demo/app.component.ts

```diff
@@ -107,6 +108,25 @@ export class AppComponent implements OnInit {
     this.form2 = this.formBuilder.group({ 'range': [ [ 2, 8 ] ] });
   }
 
+  public onSliderInit(slider: any) {
+    this.sliderObject  = slider;
+    console.debug('Slider Object Instance Initiated and available for use in your component');
+    console.debug(this.sliderObject);
+  }
+
+  public resetSlider() {
+    this.sliderObject.reset();
+  }
+
+  public togglePips() {
```

Comment:
The slider could be developed better indeed, but we are developers and we are joining forces to make that better. This pull request actually addresses a different issue, namely to provide an event when the slider is already initated, since this is not synced with angular. What you could do for now is just go with the following solution: put an element ref on your nouislider:
<code><nouislider #myslider></nouslider></code>
then in your component import ViewChild and and as class property set:
<code>@ViewChild('myslider') myslider:any</code>
then get your nouislider with:
<code>this.myslider.nativeElement.slider</code>
and then 
<code>this.myslider.nativeElement.slider.updateOptions({/*your new min max*/})</code>